src/ctia/lib/riemann.clj

```diff
@@ -0,0 +1,124 @@
+                                  {:metric ms
+                                   :tags ["ctia" "http"]
+                                   :description (str "Response took "
+                                                     (.format (java.text.DecimalFormat. "#.##")
+                                                              (/ ms 1000))
+                                                     " seconds")
+                                   :service service-name
+                                   :status (str (:status response))}))
+          response)
+        (catch ExceptionInfo e
+          (let [data (ex-data e)
+                ex-type (:type data)
+                evt {:metric (ms-elapsed start)
+                     :service (str service-name " error")
+                     :description (.getMessage e)
+                     :error "true"
+                     :status (condp = ex-type
+                               :ring.util.http-response/response (get-in data [:response :status])
+                               :compojure.api.exception/response-validation "500"
+                               :compojure.api.exception/request-validation "400"
+                               "500")}]
+            (send-request-metrics send-event-fn request evt))
+          (throw e))
+        (catch Throwable e
+          (send-request-metrics send-event-fn request
+                                {:metric (ms-elapsed start)
+                                 :service (str service-name " error")
+                                 :description (.getMessage e)
+                                 :stacktrace (.getStackTrace e)
+                                 :status "500"
+                                 :error "true"})
+          (throw e))))))
+
+(defn prepare-event [event]
+  {:user-id (get-in event [:user :id])})
+
+(defn send-event
+  [conn prefix event]
+  (let [prepared-event (-> event
+                           prepare-event
+                           (update :service #(str prefix " " %)))]
+    (if conn
+      (do
+        (log/debugf "Sending event to riemann:\n%s\nprepared:\n%s"
+                    (utils/safe-pprint event)
+                    (utils/safe-pprint prepared-event))
+        (riemann/send-event conn prepared-event))
+      (when-not (= "log" (:event-type prepared-event))
+        (log/warnf "Riemann doesn't seem configured. Event: %s"
+                   (utils/safe-pprint prepared-event))))))
+
+(defn wrap-request-logs
+  "Middleware to log all incoming connections to Riemann"
+  [service-name]
+  (let [{enabled? :enabled :as config}
+        (get-in @prop/properties [:ctia :log :riemann])]
+    (if-not enabled?
+      identity
+      (let [_ (log/info "riemann metrics reporting")
+            send-event-fn 
+            (let [client (riemann/tcp-client
+                           (select-keys config
+                                        [:host :port :interval-in-ms]))]
```

Comment:
You should probably don't want to query the config for each call.
You should probably get the client in a let before returning the function.

Reply:
I think I already return a closure that refers to the client, right?

Reply:
Yes you're right. Sorry my bad.